src/main/java/org/junitpioneer/jupiter/json/JsonConverterProvider.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.json;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.platform.commons.PreconditionViolationException;
+
+class JsonConverterProvider {
+
+	private static final boolean jacksonPresent = isClassPresent("com.fasterxml.jackson.databind.ObjectMapper");
+
+	static boolean isClassPresent(String className) {
+		try {
+			JsonConverterProvider.class.getClassLoader().loadClass(className);
+			return true;
+		}
+		catch (Throwable e) {
+			return false;
+		}
+	}
+
+	static JsonConverter getJsonConverter(ExtensionContext context) {
+		if (jacksonPresent) {
+			return JacksonJsonConverter.getConverter();
+		}
+
+		throw new PreconditionViolationException("There is no available Json parsing library");
```

Comment:
This error case comes up when users haven't configured a (supported) JSON parse, so we need to make sure, it's visible. Both Maven and Gradle will cut the error message very short, so it's unclear what the problem is.

Some options:

* throw `NoJsonParserConfiguredException` (extends `PreconditionViolationException`)
* print to `System.err`
* log a report entry with JUnit Jupiter

We should also mention which parsers we support.

Reply:
Which Maven did you use? Using 3.8.4 I get

>[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.031 s <<< FAILURE! - in org.junitpioneer.demo.JsonSourceTests
[ERROR] org.junitpioneer.demo.JsonSourceTests.jsonTest(String, int)  Time elapsed: 0.001 s  <<< ERROR!
org.junit.platform.commons.PreconditionViolationException: There is no available Json parsing library

Nothing is cut short. I will add more information about which json parsers are currently supported. 

For gradle when I do `./gradlew test` I only get that test failed. In the test report you can see the full error with its stacktrace.

When I add

```
tasks.named<Test>("test") {
	useJUnitPlatform()

	testLogging {
		setExceptionFormat("full")
	}
}
```

then I get

>JsonSourceTests > jsonTest(String, int) > org.junitpioneer.demo.JsonSourceTests.initializationError FAILED
    org.junit.platform.commons.PreconditionViolationException: There is no available Json parsing library
        at app//org.junitpioneer.jupiter.json.JsonConverterProvider.getJsonConverter(JsonConverterProvider.java:35)
        .... There is actually more stacktrace


Regarding the options you mentioned:

* What is the benefit of throwing `NoJsonParserConfiguredException`?
* I am not a fan of printing to `System.err`. So will leave it to you to add it in case you really really want it
* I can log a report entry with JUnit Jupiter, need to see how to do that first.

Reply:
I don't recall the details any more (stupid me, I should've added what I saw as text or screenshot), so I'll rely on your analysis. Regarding the options:

* The exception name showed up prominently - if that's `NoJsonParserConfiguredException`, it might not even be necessary to read the text.
* Me neither, let's not do it then.
* you know what, ignore that. I think it makes sense to limit report entries to (mostly) successful operations, not for situations that abort the test run.